### PR TITLE
drivers: gpio: add gpio controller support for max22017 dac

### PR DESCRIPTION
**Pull request by granquet**
This adds the GPIO features support to the max22017 dac device.

This depends on https://github.com/zephyrproject-rtos/zephyr/pull/76017

### CODE REVIEW

File: drivers/dac/dac_max22017.c [hunk 1]
```diff
@@ -448,11 +459,298 @@ static void max22017_int_worker(struct k_work *work)
...
+	ret = k_work_submit(&data->int_work);
+	if (ret < 0) {
+		LOG_WRN("Could not submit int work: %d", ret);
+	}
+
+	k_sem_give(&data->lock);
+}
+#endif
+
+int adi_max22017_gpio_set_output(const struct device *dev, uint8_t pin, bool initial_value)
```
Comment (martinjaeger) [inline drivers/dac/dac_max22017.c:502]: Why do all these GPIO related functions need to live in this file? Can they not be moved to `gpio_max22017.c`?

Comment (granquet) [reply drivers/dac/dac_max22017.c:502]: they can live in `gpio_max22017.c` I guess with a bit of re-org.

My rationale was that the `gpio` driver is actually just a "proxy" to the parent dac device where everything happens.
So the spi and all internal data structure of the dac driver are used to perform the gpio related functions.

Comment (henrikbrixandersen) [reply drivers/dac/dac_max22017.c:502]: Have you considered implementing an MFD base driver instead, having the DAC and GPIO drivers as children/consumers?

Comment (granquet) [reply drivers/dac/dac_max22017.c:502]: @henrikbrixandersen : The current design have been "inspired" by `drivers/adc/adc_ads114s0x.c` 
I've seen the MFD subsystem afterwards and wasn't sure it was worth using it.

I can change the design if you think using an MFD is a better choice.

Comment (martinjaeger) [reply drivers/dac/dac_max22017.c:502]: I agree that the MFD approach is better.

Comment (granquet) [reply drivers/dac/dac_max22017.c:502]: I'm looking into reworking this into an MFD then.
Dumb question: how many PRs would you like to see? 3 ? one for MFD, one for DAC, one for GPIO ?

Comment (henrikbrixandersen) [reply drivers/dac/dac_max22017.c:502]: In this case, just one.

**Comment by granquet** [comment]
design changed to use an MFD, this PR is not relevant anymore as MFD/DAC and GPIO support is now in PR https://github.com/zephyrproject-rtos/zephyr/pull/76017

**Pull request closed without merging**